db/model/model.js

```diff
@@ -8,57 +8,129 @@ exports.fetchTopics = () => {
 };
 
 exports.fetchArcticlesById = (id) => {
-  return (
-    db
-      .query(
-        `SELECT articles.*, (SELECT COUNT(*)::INT
+  return db
+    .query(
+      `SELECT DISTINCT articles.*, (SELECT COUNT(*)::INT
             FROM comments WHERE article_id=$1) AS comment_count
             FROM articles
             LEFT JOIN comments ON articles.article_id=comments.article_id WHERE articles.article_id=$1`,
-        [id]
-      )
-      .then(({ rows: [rows] }) => {
-        console.log(rows)
-        if (rows === undefined) {
-          return Promise.reject({ status: 400, msg: "Id not found" });
-        }
-        return rows;
-      })
-  );
+      [id]
+    )
+    .then(({ rows: [rows] }) => {
+      if (rows === undefined) {
+        return Promise.reject({ status: 404, msg: "Id not found" });
+      }
+      return rows;
+    });
 };
 
-exports.fetchAllArticles = (topicQuery) => {
-  console.log(topicQuery)
-  if (topicQuery === undefined) {
+exports.fetchCommentsByArticleId = (id) => {
+  return db
+    .query(
+      `SELECT *
+      FROM comments
+      WHERE article_id=$1
+      ORDER BY created_at DESC`,
+      [id]
+    )
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "Id not found" });
+      }
+      return rows;
+    });
+};
+
+exports.addNewComment = (id, newComment) => {
+  const { username, body } = newComment;
+  if (!body) {
+    return Promise.reject({ status: 400, msg: "no body" });
+  }
+  if (!username) {
+    return Promise.reject({ status: 400, msg: "no username" });
+  }  
+  return db
+    .query(
+      `INSERT INTO comments (article_id, author, body) VALUES ($1, $2, $3) RETURNING *`,
+      [id, newComment.username, newComment.body]
+    )
+    .then(({ rows: [comment] }) => {
+      return comment;
+    });
+};
+
+exports.fetchAllArticles = (topicQuery, sortByQuery, orderQuery) => {
+  if (topicQuery) {
```

Comment:
We should be able to combine the topic filter with the sort_by and order queries but we would be unable to do this with your current logic.